feat: add C implementation for `stats/base/dists/uniform/mgf` (#4142)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/uniform/mgf/README.md

<section class="references">

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/uniform/mgf.h"
```

#### stdlib_base_dists_uniform_mgf( t, a, b )

> Evaluating the logarithm of the [moment-generating function][mgf] of a [uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).

Reviewer: ```suggestion
Evaluates the [moment-generating function][mgf] (MGF) for a [continuous uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```
